### PR TITLE
Iss122 - fix rotating through mastered spells

### PR DESCRIPTION
**Pull request by Azami7**
Changed how we rotate through spells so that, rather than sneak-click, holding the wand in the off hand and right or left clicking rotates through. This keeps wand in primary hand actions consistent as always resulting in a magical action/effect happening.  

Refactored onPlayerInteract() in to helper functions since it was getting unruly.

### CODE REVIEW

File: Ollivanders/src/net/pottercraft/Ollivanders2/O2Player.java [hunk 1]
```diff
@@ -712,17 +722,25 @@ public void shiftMasterSpell ()
          }
          else
          {
-            int i = masteredSpells.indexOf(masterSpell);
+            int curSpellIndex = masteredSpells.indexOf(masterSpell);
+            int nextSpellIndex;
+
+            if (reverse)
```
Comment (autumnwoz) [inline Ollivanders/src/net/pottercraft/Ollivanders2/O2Player.java]: I think this should have brackets for consistent code style.

File: Ollivanders/src/net/pottercraft/Ollivanders2/OllivandersListener.java [hunk 2]
```diff
@@ -662,128 +709,102 @@ public void onPlayerInteract (PlayerInteractEvent event)
...
 
-               wandC = 10;
-            }
+            castSpell(player);
+            return;
+         }
+         /**
+          * A right click of the primary hand is used:
+          *  - to determine if the wand is the player's destined wand
+          *  - to set the mastered spell in the wand for non-verbal casting if they are sneaking while clicking
```
Comment (autumnwoz) [inline Ollivanders/src/net/pottercraft/Ollivanders2/OllivandersListener.java]: This isn't true anymore right?

File: Ollivanders/src/net/pottercraft/Ollivanders2/OllivandersListener.java [hunk 3]
```diff
@@ -662,128 +709,102 @@ public void onPlayerInteract (PlayerInteractEvent event)
...
+         else if (action == Action.RIGHT_CLICK_AIR || action == Action.RIGHT_CLICK_BLOCK)
+         {
+            if (Ollivanders2.debug)
+               p.getLogger().info("OllivandersListener:onPlayerInteract: right click action");
 
-            createSpellProjectile(player, spell, wandC);
-            o2p.setSpellRecentCastTime(spell);
-            int spellc = p.getSpellNum(player, spell);
-            if (spellc < 100 || spell == Spells.AVADA_KEDAVRA || !playerHoldsWand)
+            Block cauldron = (playerFacingCauldron(player));
```
Comment (autumnwoz) [inline Ollivanders/src/net/pottercraft/Ollivanders2/OllivandersListener.java]: Is this old or new potion stuff?

Comment (Azami7) [reply Ollivanders/src/net/pottercraft/Ollivanders2/OllivandersListener.java]: I think this was in my previous commit and somehow got pulled in to this PR.

**Pull request merged**